### PR TITLE
Add app context and revamps App.tsx

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Add a High Level context (TBD the content of the context itself) to load before dashboard init.
Also added some utils to make sure we persist some of the data.

## How Has This Been Tested?

Change the harcoded values from the customHook to get different screens.

## Merge criteria:

- [X] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by lucferbux** [comment]
@Griffin-Sullivan I wanna merge your PR first so then i can refactor the imports to the right format.
Also, I'm gonna add the first cypress tests to cover the loading and error state, will do once merged the other PR.

File: clients/ui/frontend/src/app/App.tsx [hunk 1]
```diff
@@ -1,20 +1,79 @@
...
+    // There was an error fetching critical data
+    return (
+      <Page>
+        <PageSection>
+          <Stack hasGutter>
+            <StackItem>
+              <Alert variant="danger" isInline title="General loading error">
+                <p>
+                  {(userError ? "User not loaded" : "Config not loaded") ||
+                    "Unknown error occurred during startup."}{" "}
```
Comment (Griffin-Sullivan) [inline clients/ui/frontend/src/app/App.tsx:52]: What's the purpose of the empty space? `{" "}`

File: clients/ui/frontend/src/app/App.tsx [hunk 2]
```diff
@@ -1,20 +1,79 @@
...
+                  // TODO: replace with actual error message
```
Comment (Griffin-Sullivan) [inline clients/ui/frontend/src/app/App.tsx:53]: Not sure if you meant for this to render, but it's actually showing up in the UI. Based off the comment syntax I imagine you want this to be in JSX

**Review by Griffin-Sullivan — COMMENTED** [review]
Couple of small things that don't matter. I tested the different states and they loaded properly. userError = true, configError = true, loading = true, and normal. Not sure how we can test browser storage at the moment so can't speak on that, but the code looks good to me.

File: clients/ui/frontend/src/app/App.tsx (hunk 2 above)
Comment (lucferbux) [reply clients/ui/frontend/src/app/App.tsx]: Yeah, that was just to get an error in the future, but I've refactored it, better to implement the customHook with real values and just refactor that.

File: clients/ui/frontend/src/app/App.tsx [hunk 3]
```diff
@@ -3,22 +3,88 @@ import '@patternfly/react-core/dist/styles/base.css';
...
+} from "@patternfly/react-core";
+import NavSidebar from "./NavSidebar";
+import { BarsIcon } from "@patternfly/react-icons";
+import { AppContext } from "./AppContext";
+import { useSettings } from "./useSettings";
 
 const App: React.FC = () => {
+  const {
+    configSettings,
+    userSerttings,
```
Comment (Griffin-Sullivan) [inline clients/ui/frontend/src/app/App.tsx:29]:
```suggestion
    userSettings,
```

File: clients/ui/frontend/src/app/useSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import * as React from "react";
+import useTimeBasedRefresh from "./useTimeBasedRefresh";
+import { ConfigSettings, PlatformType, UserSettings } from "../types";
+import { POLL_INTERVAL } from "~/utilities/const";
+import { useDeepCompareMemoize } from "../utilities/useDeepCompareMemoize";
+
+export const useSettings = (): {
+  configSettings: ConfigSettings | null;
+  userSerttings: UserSettings | null;
```
Comment (Griffin-Sullivan) [inline clients/ui/frontend/src/app/useSettings.tsx:9]:
```suggestion
  userSettings: UserSettings | null;
```

File: clients/ui/frontend/src/app/App.tsx (hunk 3 above)
Comment (lucferbux) [reply clients/ui/frontend/src/app/App.tsx]: Good catch!

**Review by Griffin-Sullivan — APPROVED** [review]

**Comment by alexcreasy** [comment]
@lucferbux just needs the test failure fixed

**Review by ederign — CHANGES_REQUESTED** [review]
Hold on this until we discuss a few things!

**Review by ederign — APPROVED** [review]

**Pull request merged**